OpenRA.Mods.Common/AI/States/GroundStates.cs

```diff
@@ -130,19 +140,21 @@ public void Tick(Squad owner)
 
 			if (!owner.IsTargetValid)
 			{
-				var closestEnemy = owner.Bot.FindClosestEnemy(owner.Units.Random(owner.Random).CenterPosition);
-				if (closestEnemy != null)
-					owner.TargetActor = closestEnemy;
+				var t = FindClosestEnemy(owner);
+				if (t != null)
+					owner.TargetActor = t;
 				else
 				{
 					owner.FuzzyStateMachine.ChangeState(owner, new GroundUnitsFleeState(), true);
 					return;
 				}
 			}
 
+			// Switch target durign fight
+			var targetActor = owner.Bot.FindClosestEnemy(owner.Units.First().CenterPosition);
 			foreach (var a in owner.Units)
 				if (!BusyAttack(a))
-					owner.Bot.QueueOrder(new Order("Attack", a, false) { TargetActor = owner.Bot.FindClosestEnemy(a.CenterPosition) });
+					owner.Bot.QueueOrder(new Order("Attack", a, false) { TargetActor = targetActor });
```

Comment:
In a quick test game, looks like orders can very often be issued in bulk, sometimes dozens at once. I'd say that makes this very worthwhile. Nice spot.